contracts/ReversibleICO.sol

```diff
@@ -322,13 +357,19 @@ contract ReversibleICO is IERC777Recipient {
     {
         require(msg.sender == projectWalletAddress, "Only project wallet address.");
 
+        // get project unlocked ETH (available for withdrawing)
         uint256 unlocked = getProjectAvailableEth();
-        require(_ethAmount <= unlocked, "Requested amount to large, not enough unlocked ETH available.");
+        require(_ethAmount <= unlocked, "Requested amount too big, not enough unlocked ETH available.");
 
+        // update stats:  #project withdrawals, total amount withdrawn by the project
         projectWithdrawCount++;
         projectWithdrawnETH += _ethAmount;
 
 
+        // transfer ETH to project wallet
+        address(uint160(projectWalletAddress)).transfer(_ethAmount);
```

Comment:
Strange why you moved the transfer before the events,
Is there some kind of standard that says "events should be emitted" at the end of an operation ?

The only thing that comes to mind that might be affected by emitting and transferring later is resulting event order in like an app ( web3 filter ) on the resulting block.

Reply:
There is no standard or serious reason, it's mainly a matter of coding conventions, the only thing is that if the transfer reverts then the whole thing will revert saving a bit of gas. I think it better to execute first the statements that might revert (depending on the use case of course.